js/main.js

```diff
@@ -23,4 +52,51 @@ const randomCoords = (min, max, decimalPoint) => {
   throw new Error('Диапазон значений может быть только положительный, включая ноль');
 };
 
-randomCoords(1.1, 1.2, 3);
+const getRandomArrayElement = function (array) {
+  return array[randomInt(0, array.length - 1)];
+};
+
+const getRandomArray = (array) => {
+  const randomArray = [];
+  for (let counter = 0; counter < array.length; counter++) {
+    if (Math.random() > 0.5) {
+      randomArray.push(array[counter]);
+    }
+  }
+  return randomArray;
+};
+
+const createAdvertisement = () => {
+  const LOCATION = {
+    lat: randomCoords(35.65, 35.7, 4),
+    lng: randomCoords(139.7, 139.8, 4),
+  };
+
+  const getLocation = (location) => `${location.lat}, ${location.lng}`;
+
+  const AUTHOR = {
+    avatar: `img/avatars/user0${randomInt(1, 8)}.png`,
+  };
+
+  const OFFER = {
+    title: 'Самые бронируемые апартаменты/квартиры в этом месяце',
+    address: getLocation(LOCATION),
+    price: randomInt(1, Number.MAX_SAFE_INTEGER),
+    type: getRandomArrayElement(TYPE_OF_LIVING_ACCOMODATION),
+    rooms: randomInt(1, Number.MAX_SAFE_INTEGER),
+    guests: randomInt(1, Number.MAX_SAFE_INTEGER),
+    checkin: getRandomArrayElement(CHECK_IN_OUT_TIME),
+    checkout: getRandomArrayElement(CHECK_IN_OUT_TIME),
+    features: getRandomArray(FEATURES),
+    description: 'Космополитический 5-звездочный отель Pullman Dubai Downtown расположен в районе Бизнес-Бэй, недалеко от небоскреба Бурдж-Халифа, торгового центра Dubai Mall и района Даунтаун-Дубай.',
+    photos: getRandomArray(PHOTOS),
+  };
+  return { AUTHOR, LOCATION, OFFER };
+};
+const similarAdvertisement = new Array(COUNT_OF_ADVERTISEMENT).fill(null).map(() => createAdvertisement());
```

Comment:
Для именования массивов используется множественное число, [критерий Б5](https://up.htmlacademy.ru/javascript/23/criteries#b5).
```suggestion
const similarAdvertisements = new Array(COUNT_OF_ADVERTISEMENT).fill(null).map(() => createAdvertisement());
```